sql/src/main/java/io/crate/analyze/CreateSnapshotAnalyzer.java

```diff
@@ -103,12 +104,7 @@ public CreateSnapshotAnalyzedStatement analyze(CreateSnapshot node, Analysis ana
                         throw e;
                     }
                 }
-                if (!(tableInfo instanceof DocTableInfo)) {
-                    throw new IllegalArgumentException(
-                        String.format(Locale.ENGLISH, "Cannot create snapshot of tables in schema '%s'", tableInfo.ident().schema()));
-                }
-                Operation.blockedRaiseException(tableInfo, Operation.READ);
-                DocTableInfo docTableInfo = (DocTableInfo) tableInfo;
+                Operation.blockedRaiseException(docTableInfo, Operation.READ);
```

Comment:
afaik this can be removed, it is (or should) already be handled while resolving the tableInfo using the `Operation.CREATE_SNAPSHOT`